Sources/KeyboardLayoutGuide/KeyboardLayoutGuide.swift

```diff
@@ -112,6 +112,26 @@ open class KeyboardLayoutGuide: UILayoutGuide {
             if #available(iOS 11.0, *), usesSafeArea, height > 0, let bottom = owningView?.safeAreaInsets.bottom {
                 height -= bottom
             }
+
+            // Factor in the non full screen views (iPad sheets)
+            if let activeWindow = UIApplication.shared.activeWindow, let owningView = owningView, height > 0 {
+                // Owning view's frame in the active window
+                let owningFrameInRoot = owningView.convert(owningView.frame, to: activeWindow)
+                // We have to ignore the part of the owning view that's outside of the active window
+                // (Modal presentation)
+                let intersectionFrame = activeWindow.frame.intersection(owningFrameInRoot)
+
+                let windowHeight = activeWindow.frame.height
+                let bottomDifference = windowHeight - (intersectionFrame.height + intersectionFrame.origin.y)
```

Comment:
I think you can use `intersectionFrame.maxY` here?

Reply:
You are right, thanks.